static/scripts/rewards/web3/connect-wallet.ts

```diff
@@ -12,18 +13,74 @@ export async function connectWallet(): Promise<JsonRpcSigner | null> {
 
     const address = await signer.getAddress();
 
+    const provider = new ethers.providers.JsonRpcProvider(wallet.provider.url);
+
     if (!address) {
       buttonController.hideAll();
       console.error("Wallet not connected");
       return null;
     }
 
+    const isOkay = await stressTestWalletRpc(provider);
+
+    if (!isOkay) {
+      toaster.create("error", "We have detected potential issues with your in-wallet RPC. Accept the request to replace it with a more reliable one.");
+      await addFastestHandlerNetwork();
+      return null;
+    }
+
     return signer;
   } catch (error: unknown) {
     return connectErrorHandler(error);
   }
 }
 
+async function addFastestHandlerNetwork() {
+  const provider = await useRpcHandler(app.networkId ?? provider.network.chainId);
+  const url = provider.connection.url;
+
+  try {
+    await provider.request({
+      method: "wallet_addEthereumChain",
+      params: [
+        {
+          chainId: app.networkId,
+          chainName: url,
+          rpcUrls: [url],
+        },
+      ],
+    });
+  } catch (error) {
+    console.error(error);
+    toaster.create("error"`Failed to add optimal RPC network. Please add it manually. Network ID: ${app.networkId}, URL: ${url}`);
+  }
+}
+
+async function stressTestWalletRpc(provider: ethers.providers.Web3Provider) {
+  const success = [];
+
+  for (let i = 0; i < 10; i++) {
+    success.push(await testEthCall(provider));
+  }
+
+  return success.filter((s) => s === "0x" + "00".repeat(32)).length > 9;
+}
+
+async function testEthCall(provider: ethers.providers.Web3Provider) {
+  try {
+    return await provider.send("eth_call", [
+      {
+        to: "0x000000000022D473030F116dDEE9F6B43aC78BA3",
+        input: "0x4fe02b44000000000000000000000000d9530f3fbbea11bed01dc09e79318f2f20223716001fd097bcb5a1759ce02c0a671386a0bbbfa8216559e5855698a9d4de4cddea",
```

Comment:
What's the input?

Reply:
It's a fake permit from the V2 bot testing for me, so it's expected that the result will always be the same unless for some reason all previous V2 bot permits get invalidated

Reply:
I've made it clearer as to what it is